pkg/apis/certificates/validation/validation.go

```diff
@@ -572,3 +580,256 @@ func validateTrustBundle(path *field.Path, in string) field.ErrorList {
+func ValidatePodCertificateRequest(req *certificates.PodCertificateRequest, opts ValidatePodCertificateRequestOptions) field.ErrorList {
+	var allErrors field.ErrorList
+
+	metaErrors := apivalidation.ValidateObjectMeta(&req.ObjectMeta, true, validatePodCertificateRequestName, field.NewPath("metadata"))
+	allErrors = append(allErrors, metaErrors...)
+
+	signerNameErrors := apivalidation.ValidateSignerName(field.NewPath("spec", "signerName"), req.Spec.SignerName)
+	allErrors = append(allErrors, signerNameErrors...)
+
+	// TODO(KEP-4317): Basic format validation on other spec fields.
+
+	// TODO(KEP-4317): Denied / Failed conditions are not allowed if the
+	// certificate is issued.
+
+	// Everything below this point is crypto verification that should be skipped
+	// if the fields haven't changed.
+	if opts.SkipCryptoValidation {
+		return allErrors
+	}
+
+	// TODO(KEP-4317): Check length of PKIXPublicKey and ProofOfPossession
+	// fields.
+
+	pubAny, err := x509.ParsePKIXPublicKey(req.Spec.PKIXPublicKey)
+	if err != nil {
+		allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "must be a valid PKIX-serialized public key"))
+		return allErrors
+	}
+
+	// Verify public key properties and the proof-of-possession signature.
+
+	switch pub := pubAny.(type) {
+	case ed25519.PublicKey:
+		// ed25519 has no key configuration to check
+		if !ed25519.Verify(pub, []byte(req.Spec.PodUID), req.Spec.ProofOfPossession) {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	case *ecdsa.PublicKey:
+		if pub.Curve != elliptic.P256() && pub.Curve != elliptic.P384() {
+			allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "elliptic public keys must use curve P256 or P384"))
+			return allErrors
+		}
+		if !ecdsa.VerifyASN1(pub, hashBytes([]byte(req.Spec.PodUID)), req.Spec.ProofOfPossession) {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	case *rsa.PublicKey:
+		if pub.Size() != 3072 && pub.Size() != 4096 {
+			allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "RSA keys must have modulus size 3072 or 4096"))
+			return allErrors
+		}
+		if err := rsa.VerifyPKCS1v15(pub, crypto.SHA256, hashBytes([]byte(req.Spec.PodUID)), req.Spec.ProofOfPossession); err != nil {
+			allErrors = append(allErrors, field.Invalid(popPath, req.Spec.ProofOfPossession, "could not verify proof-of-possession signature"))
+			return allErrors
+		}
+
+	default:
+		allErrors = append(allErrors, field.Invalid(pkixPath, req.Spec.PKIXPublicKey, "unknown public key type; supported types are Ed25519, ECDSA, and RSA"))
+		return allErrors
+	}
+
+	// Bail out if the certificate chain hasn't been set.
+	if req.Status.CertificateChain == "" {
```

Comment:
.status updates need to be a different function.  .status updates must never fail on invalid .spec values.

Reply:
Done.